contrib/linux-agent-installer/Icinga2Agent.bash

```diff
@@ -1,77 +1,202 @@
-ICINGA2_CONF_DIR=$(dirname "$ICINGA2_CONFIG_FILE")
-ICINGA2_SYSCONF_DIR=$(dirname "$ICINGA2_CONF_DIR")
-ICINGA2_INSTALL_PREFIX=$(dirname $(dirname "$DAEMON"))
-ICINGA2_CA_DIR="${ICINGA2_STATE_DIR}/lib/icinga2/ca"
-ICINGA2_SSL_DIR="${ICINGA2_CONF_DIR}/pki"
-ICINGA2_CA_PORT="5665"
+# internal defaults
+: "${ICINGA2_CONFIG_FILE:=/etc/icinga2/icinga2.conf}"
+: "${ICINGA2_CONFIGDIR:="$(dirname "$ICINGA2_CONFIG_FILE")"}"
+: "${ICINGA2_DATADIR:=/var/lib/icinga2}"
+: "${ICINGA2_SSLDIR_OLD:="${ICINGA2_CONFIGDIR}"/pki}"
+: "${ICINGA2_SSLDIR_NEW:="${ICINGA2_DATADIR}"/certs}"
+: "${ICINGA2_SSLDIR:=}"
+: "${ICINGA2_BIN:=icinga2}"
+
+case "$ICINGA2_OSFAMILY" in
+debian)
+  : "${ICINGA2_USER:=nagios}"
+  : "${ICINGA2_GROUP:=nagios}"
+  ;;
+redhat)
+  : "${ICINGA2_USER:=icinga}"
+  : "${ICINGA2_GROUP:=icinga}"
+  ;;
+*)
+  fail "Unknown osfamily '$ICINGA2_OSFAMILY'!"
+  ;;
+esac
+
+icinga_version() {
+  "$ICINGA2_BIN" --version 2>/dev/null | grep -oP 'version: [rv]\K\d+\.\d+\.\d+[^\)]*'
+}
+
+icinga_major() {
+  icinga_version | grep -oP '^\d+\.\d+'
+}
 
-. "${ICINGA2_INSTALL_PREFIX}/lib/icinga2/prepare-dirs" "${ICINGA2_SYSCONFIG_FILE}"
+# only can do float like versions (1.x)
+version_compare() {
+  # args: v1 op v2
+  return "$(awk "BEGIN {print !($1 $2 $3)}")"
+}
 
-if ! [ -d $ICINGA2_SSL_DIR ]; then mkdir $ICINGA2_SSL_DIR; fi
-chown $ICINGA2_USER $ICINGA2_SSL_DIR
+# Make sure icinga2 is installed and running
+echo -n "check: icinga2 installed - "
+if version=$(icinga_version); then
+  echo "OK: $version"
+else
+  fail "You need to install icinga2!"
+fi
 
-if [ -f  "${ICINGA2_SSL_DIR}/${ICINGA2_NODENAME}.crt" ]; then
+if [ -z "${ICINGA2_SSLDIR}" ]; then
+  if [ -f "${ICINGA2_SSLDIR_OLD}/${ICINGA2_NODENAME}.crt" ]; then
+    info "Using old SSL directory: ${ICINGA2_SSLDIR_OLD}"
+    info "Because you already have a certificate in ${ICINGA2_SSLDIR_OLD}/${ICINGA2_NODENAME}.crt"
+    ICINGA2_SSLDIR="${ICINGA2_SSLDIR_OLD}"
+  elif version_compare "$(icinga_major)" ">=" 2.8 ; then
+    info "Using new SSL directory: ${ICINGA2_SSLDIR_NEW}"
+    ICINGA2_SSLDIR="${ICINGA2_SSLDIR_NEW}"
+  else
+    info "Using old SSL directory: ${ICINGA2_SSLDIR_OLD}"
+    ICINGA2_SSLDIR="${ICINGA2_SSLDIR_OLD}"
+  fi
+fi
+
+if [ ! -d "$ICINGA2_SSLDIR" ]; then
+  mkdir "$ICINGA2_SSLDIR"
+  chown "$ICINGA2_USER.$ICINGA2_GROUP" "$ICINGA2_SSLDIR"
+fi
+
+if [ -f "${ICINGA2_SSLDIR}/${ICINGA2_NODENAME}.crt" ]; then
   warn "ERROR: a certificate for '${ICINGA2_NODENAME}' already exists"
-  warn "Please remove ${ICINGA2_SSL_DIR}/${ICINGA2_NODENAME}.??? in case you want a"
+  warn "Please remove ${ICINGA2_SSLDIR}/${ICINGA2_NODENAME}.??? in case you want a"
   warn "new certificate to be generated and signed by ${ICINGA2_CA_NODE}"
-  exit 1
+
+  if [ -z "${ICINGA2_UPDATE_CONFIG}" ] && [ -z "${ICINGA2_DRYRUN}" ]; then
+    warn "Aborting here, you can can call the script like this to just update config:"
+    info " ICINGA2_UPDATE_CONFIG=1 $0"
```

Comment:
:sob:

Reply:
Why cry?^^

Reply:
`PREFIXED_LONG_ARGUMENT=1 ./run-command`

VS

`./run-command --long-argument`

I understand that this was easier to implement, but it feels dirty. Merged it anyways, don't worry.